lib/gems/pending/util/object_storage/miq_ftp_storage.rb

```diff
@@ -0,0 +1,92 @@
+require 'util/miq_ftp_lib'
+require 'util/miq_object_storage'
+require 'logger'
+
+class MiqFtpStorage < MiqObjectStorage
+  include MiqFtpLib
+
+  FTP_CHUNKSIZE = ::Net::FTP::DEFAULT_BLOCKSIZE
+
+  attr_reader :uri, :username, :password
+
+  class << self
+    attr_writer :instance_logger
+
+    def instance_logger
+      @instance_logger ||= $log || Logger.new(File::NULL)
+    end
+  end
+
+  def self.uri_scheme
+    "ftp".freeze
+  end
+
+  def initialize(settings)
+    super
+    @uri      = @settings[:uri]
+    @username = @settings[:username]
+    @password = @settings[:password]
+  end
+
+  # Override for connection handling
+  def add(*upload_args)
+    with_connection { super }
+  end
+
+  # Override for connection handling
+  def download(*upload_args)
```

Comment:
Probably should be called `download_args`

Reply:
(－‸ლ)

I'm a derp... (will fix)